mission_3/routes/products.js

```diff
@@ -0,0 +1,335 @@
+    } = req.query;
+
+    let productsSort = '';
+
+    switch (sort) {
+      case 'old':
+        productsSort = 'asc';
+        break;
+
+      case 'recent':
+      default:
+        productsSort = 'desc';
+    }
+
+    try {
+      const products = await prisma.product.findMany({
+        select: {
+          id: true,
+          name: true,
+          price: true,
+          createdAt: true
+        },
+        orderBy: {
+          createdAt: productsSort
+        },
+        where: {
+          OR: [
+            {
+              name: {
+                contains: search,
+                mode: 'insensitive',
+             }
+            },
+            {
+              description: {
+                contains: search,
+                mode: 'insensitive',
+              }
+            }
+          ],
+        },
+        skip: offset * limit,
+        take: limit
+      });
+
+      if (products)
+        res.status(200).json(products);
+      else 
+        res.status(404).json({ message: `Cannot find product with ID ${id}` });
+    } catch (err) {
+      console.error('An error has occurred: ', err.message);
+
+      res.status(500).json({ message: "An error has occurred during processing sql" });
+    }
+  });
+
+
+
+router.route('/:id')
+
+  // Get informations of a certain product
+  .get(async (req, res) => {
+    const { id } = req.params;
+    if (!id)
+      return res.status(400).json({ message: "Invalid parameter 'id'" });
+
+    try {
+      const product = await prisma.product.findUnique({
+        where: {
+          id: parseInt(id)
+        }
+      });
+
+      if (product)
+        res.status(200).json(product);
+      else 
+        res.status(404).json({ message: `Cannot find product with ID ${id}` })
+    } catch (err) {
+      console.error('An error has occurred: ', err.message);
+
+      res.status(500).json({ message: "An error has occurred during processing sql" });
+    }
+  })
+
+  // Modify a product property
+  .patch(async (req, res) => {
+    const { id } = req.params;
+    if (!id)
+      return res.status(400).json({ message: "Invalid parameter 'id'" });
+
+    // Columns in model Product
+    const productCols = [
+      "name",
+      "description",
+      "price",
+      "tags"
+    ];
+
+    const filteredBody = Object.entries(req.body)
+      .filter(e => productCols.includes(e[0]))
+      .reduce((obj, ele) => {
+        obj[ele[0]] = ele[1]
+        return obj;
+      }, {});
+    
+    try {
+      const product = await prisma.product.update({
+        where: {
+          id: Number(id)
+        },
+        data: {
+          ...filteredBody
+        }
+      });
+
+      if (product)
+        res.status(200).json(product);
+      else 
+        res.status(404).json({ message: `Cannot find product with ID ${id}` });
+    } catch (err) {
+      console.error('An error has occurred: ', err.message);
+
+      res.status(500).json({ message: "An error has occurred during processing sql" });
+    }
```

Comment:
에러처리를 조금 더 개선하면 좋을 것 같습니다. prisma에서 어떤 상황일때 어떤 에러코드를 전달하고, 어떻게 에러 처리해야하는지도 공식문서를 보시면서 확인하면 좋을 것 같습니다.

```suggestion
try {
  const product = await prisma.product.update({
    where: {
      id: Number(id)
    },
    data: filteredBody
  });

  return res.status(200).json(product);
} catch (err) {
  if (err.code === 'P2025') {
    return res.status(404).json({ message: `Product with ID ${id} not found` });
  }
  console.error('An error has occurred:', err);
  return res.status(500).json({ message: "Internal server error" });
}
```